docs/sources/flow/reference/components/otelcol.processor.resourcedetection.md

```diff
@@ -0,0 +1,940 @@
+system            | [system][]            |                                                   | no
+openshift         | [openshift][]         |                                                   | no
+kubernetes_node   | [kubernetes_node][]   |                                                   | no
+
+[output]: #output
+[ec2]: #ec2
+[ecs]: #ecs
+[eks]: #eks
+[elasticbeanstalk]: #elasticbeanstalk
+[lambda]: #lambda
+[azure]: #azure
+[aks]: #aks
+[consul]: #consul
+[docker]: #docker
+[gcp]: #gcp
+[heroku]: #heroku
+[system]: #system
+[openshift]: #openshift
+[kubernetes_node]: #kubernetes_node
+
+[res-attr-cfg]: #resource-attribute-config
+
+### output
+
+{{< docs/shared lookup="flow/reference/components/output-block.md" source="agent" version="<AGENT VERSION>" >}}
+
+### ec2
+
+The `ec2` block reads resource information from the [EC2 instance metadata API] using the [AWS SDK for Go][].
+
+The `ec2` block supports the following attributes:
+
+Attribute   | Type           | Description                                                                 | Default     | Required
+----------- |----------------| --------------------------------------------------------------------------- |-------------| --------
+`tags`      | `list(string)` | A list of regular expressions to match against tag keys of an EC2 instance. | `[]`        | no
+
+If you are using a proxy server on your EC2 instance, it's important that you exempt requests for instance metadata as described in the [AWS cli user guide][]. 
+Failing to do so can result in proxied or missing instance data.
+
+If the instance is part of AWS ParallelCluster and the detector is failing to connect to the metadata server, 
+check the iptable and make sure the chain `PARALLELCLUSTER_IMDS` contains a rule that allows the {{< param "PRODUCT_ROOT_NAME" >}} user to access `169.254.169.254/32`.
+
+[AWS SDK for Go]: https://docs.aws.amazon.com/sdk-for-go/api/aws/ec2metadata/
+[EC2 instance metadata API]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+[AWS cli user guide]: https://github.com/awsdocs/aws-cli-user-guide/blob/a2393582590b64bd2a1d9978af15b350e1f9eb8e/doc_source/cli-configure-proxy.md#using-a-proxy-on-amazon-ec2-instances
+
+`tags` can be used to gather tags for the EC2 instance which {{< param "PRODUCT_ROOT_NAME" >}} is running on. 
+Note that in order to fetch EC2 tags, the IAM role assigned to the EC2 instance must have a policy that includes the `ec2:DescribeTags` permission.
+
+The `ec2` block supports the following blocks:
+
+Block                                          | Description                                       | Required
+---------------------------------------------- | ------------------------------------------------- | --------
+[resource_attributes](#ec2--resource_attributes) | Configures which resource attributes to add.      | no
+
+##### ec2 > resource_attributes
```

Comment:
These `... > resource_attributes` headings have a small weight of `####` so that they don't appear in the ToC. This makes the ToC much cleaner.